### PR TITLE
Improve push notifications

### PR DESCRIPTION
**Pull request by david-venhoff**
### Short description

This pr fixes the problems we found after the merge of the push notification features.
Before this is merged, a database migration will be required which sets `draft=False` on push notifications created before update `2023.8.0` 

### Proposed changes

- As described in the issue

### Side effects

/

### Resolved issues

Fixes: #2375 

__________________________________________________

[Pull Request Review Guidelines](https://digitalfabrik.github.io/integreat-cms/pull-request-review-guide.html)

### CODE REVIEW

**Comment by svenseeberg** [comment]
Awesome, thanks! :tada:

Just a reminder before we deploy this: we need to update the database to `draft=False` for all messages that were created before we deploy(ed) version `2023.8.0` to the Integreat and Malte CMS. Malte was deployed on Monday (2023-08-01 at 0830 CEST). The Integreat CMS will receive the update in the morning of Friday (2023-08-11).

**Comment by svenseeberg** [comment]
Generally the content of the PR looks also quite good, thanks :+1:

One bold question though: would you be willing to include the change of the checkboxes for the multi-region messages to a nice & pretty multiselect? :eyes: If you do not want to include this, we can also split this into another issue. The changes of this PR are important enough to merge them anyways :)

**Review by PeterNerlich — CHANGES_REQUESTED** [review]
Thank you for cleaning up our mess :rofl: 

I noticed the success message on creating or saving templates still says "News message … was successfully saved". Also, when scheduling a message, we still run the checks as if we were to actually send it now, in order to catch any errors that would prevent sending when the set date is reached. However, if it fails, the message to the user is still in past tense: "…*could* not be sent." I think we can change it to "…*cannot* be sent.", that will fit universally.
I suggest to include these changes as part of this PR (sadly, they don't fit in a nice *commit suggestion* box by GitHub because they affect lines not added in this PR):
```diff
diff --git a/integreat_cms/cms/views/push_notifications/push_notification_form_view.py b/integreat_cms/cms/views/push_n
otifications/push_notification_form_view.py
index eaa586303..8de96591f 100644
--- a/integreat_cms/cms/views/push_notifications/push_notification_form_view.py
+++ b/integreat_cms/cms/views/push_notifications/push_notification_form_view.py
@@ -283,19 +283,35 @@ class PushNotificationFormView(TemplateView):
 
             # Add the success message
             if not push_notification_instance:
-                messages.success(
-                    request,
-                    _('News message "{}" was successfully created').format(
-                        pn_form.instance
-                    ),
-                )
+                if pn_form.cleaned_data["is_template"]:
+                    messages.success(
+                        request,
+                        _('Template "{}" was successfully created').format(
+                            pn_form.cleaned_data["template_name"]
+                        ),
+                    )
+                else:
+                    messages.success(
+                        request,
+                        _('News message "{}" was successfully created').format(
+                            pn_form.instance
+                        ),
+                    )
             else:
-                messages.success(
-                    request,
-                    _('News message "{}" was successfully saved').format(
-                        pn_form.instance
-                    ),
-                )
+                if pn_form.cleaned_data["is_template"]:
+                    messages.success(
+                        request,
+                        _('Template "{}" was successfully saved').format(
+                            pn_form.cleaned_data["template_name"]
+                        ),
+                    )
+                else:
+                    messages.success(
+                        request,
+                        _('News message "{}" was successfully saved').format(
+                            pn_form.instance
+                        ),
+                    )
 
             # The submit_send submit button is used in 2 cases:
             # 1. send a push notification
@@ -349,16 +365,16 @@ class PushNotificationFormView(TemplateView):
                         pn_form.instance.draft = False
                         pn_form.instance.save()
                     else:
-                        messages.error(request, _("News message could not be sent"))
+                        messages.error(request, _("News message cannot be sent"))
                 except ImproperlyConfigured as e:
                     logger.error(
-                        "News message could not be sent due to a configuration error: %s",
+                        "News message cannot be sent due to a configuration error: %s",
                         e,
                     )
                     messages.error(
                         request,
                         _(
-                            "News message could not be sent due to a configuration error."
+                            "News message cannot be sent due to a configuration error."
                         ),
                     )
             else:
diff --git a/integreat_cms/locale/de/LC_MESSAGES/django.po b/integreat_cms/locale/de/LC_MESSAGES/django.po
index 8ee8df74e..b9592423a 100644
--- a/integreat_cms/locale/de/LC_MESSAGES/django.po
+++ b/integreat_cms/locale/de/LC_MESSAGES/django.po
@@ -8765,10 +8765,18 @@ msgstr ""
 "Bitte wenden Sie sich an denjenigen, der für all diese Regionen zuständig "
 "ist oder kontaktieren Sie eine:n Administrator:in."
 
+#: cms/views/push_notifications/push_notification_form_view.py
+msgid "Template \"{}\" was successfully created"
+msgstr "Vorlage \"{}\" wurde erfolgreich erstellt"
+
 #: cms/views/push_notifications/push_notification_form_view.py
 msgid "News message \"{}\" was successfully created"
 msgstr "Nachricht \"{}\" wurde erfolgreich erstellt"
 
+#: cms/views/push_notifications/push_notification_form_view.py
+msgid "Template \"{}\" was successfully saved"
+msgstr "Vorlage \"{}\" wurde erfolgreich gespeichert"
+
 #: cms/views/push_notifications/push_notification_form_view.py
 msgid "News message \"{}\" was successfully saved"
 msgstr "Nachricht \"{}\" wurde erfolgreich gespeichert"
@@ -8782,13 +8790,13 @@ msgid "News message was successfully sent"
 msgstr "Nachricht wurde erfolgreich gesendet"
 
 #: cms/views/push_notifications/push_notification_form_view.py
-msgid "News message could not be sent"
-msgstr "Nachricht konnte nicht gesendet werden"
+msgid "News message cannot be sent"
+msgstr "Nachricht kann nicht gesendet werden"
 
 #: cms/views/push_notifications/push_notification_form_view.py
-msgid "News message could not be sent due to a configuration error."
+msgid "News message cannot be sent due to a configuration error."
 msgstr ""
-"Nachricht konnte aufgrund eines Konfigurationsproblems nicht gesendet werden."
+"Nachricht kann aufgrund eines Konfigurationsproblems nicht gesendet werden."
 
 #: cms/views/push_notifications/push_notification_list_view.py
 msgid ""
```

File: integreat_cms/locale/de/LC_MESSAGES/django.po [hunk 1]
```diff
@@ -8782,13 +8790,13 @@ msgid "News message was successfully sent"
 msgstr "Nachricht wurde erfolgreich gesendet"
 
 #: cms/views/push_notifications/push_notification_form_view.py
-msgid "News message could not be sent"
-msgstr "Nachricht konnte nicht gesendet werden"
+msgid "News message cannot not be sent"
```
Comment (PeterNerlich) [inline integreat_cms/locale/de/LC_MESSAGES/django.po:8793]:
```suggestion
msgid "News message cannot be sent"
```

**Review by PeterNerlich — APPROVED** [review]
Almost, you only missed a `not`, then we're good to go imo

**Comment by david-venhoff** [comment]
Done. Is it correct to use `news` in singular form? Or should the sentences be updated to plural form?

**Comment by PeterNerlich** [comment]
> Is it correct to use `news` in singular form?

*News* is like *sand*, *data* or *information* linguistically. They *are* the plural form. Rather, there exists no singular form, since it doesn't make sense; you can use auxiliary words like *a **grain** of sand* or *a **piece** of news*, but not ~~*one sand*~~ – it's uncountable.
In German it even is kind of similar for news: Yes, the singular form *Nachricht* exists, but only with the meaning of *message*. *Nachrichten* can mean both *messages* or *news*, and is similarly uncountable in the latter case.

**Review by svenseeberg — APPROVED** [review]
Good to go, but please either rebase or squash merge :)

**Pull request merged**